environment/lattice.py

```
@@ -320,32 +342,53 @@ def generate_outer_update(self, now):
                 location = self.locations[agent_id][0:2] * self.patches_per_edge / self.edge_length
                 patch_site = tuple(np.floor(location).astype(int))
                 update[agent_id] = {}
-                update[agent_id]['concentrations'] = dict(zip(
-                    self._molecule_ids,
-                    self.lattice[:, patch_site[0], patch_site[1]]))
+                try:
+                    update[agent_id]['concentrations'] = dict(zip(
+                        self._molecule_ids,
+                        self.lattice[:, patch_site[0], patch_site[1]]))
+                except:
```

Careful. A bare `except:` clause is OK only when printing/logging then re-raising the traceback, otherwise it blocks SystemExit, StopIteration, KeyboardInterrupt, etc. `except StandardError:` is better but that will still swallow up runtime errors like NameError and AttributeError, i.e. if the code has a misspelled method or class name. I wish there was a good exception category for ordinary errors like ValueError et al.

(See https://github.com/CovertLab/wcEcoli/blob/master/docs/style-guide.md)

---

I changed this from a ```try```/```except``` to an ```assert```:

```assert (0 <= patch_site[0] < self.patches_per_edge)```
```assert (0 <= patch_site[1] < self.patches_per_edge)```

---

this shouldn't be a problem. It came up while I was integrating Pymunk, and the elasticity of bodies pushed up against the boundary made it so some cells appeared to be crossing the boundary.